flow_screen_components/quickChoiceFSCproject/force-app/main/default/lwc/quickChoiceFSC/quickChoiceFSC.js

```diff
@@ -1,167 +1,215 @@
+		console.log("objectFieldName is: " + this.objectAndFieldName);
+		if (!this.recordTypeId) this.recordTypeId = this.masterRecordTypeId;
+
+		// Visual Picker Selection
+		if (this.displayMode === "Visual") {
+			this.showVisual = true;
+			this.showRadio = false;
+			console.log("showIcons is: " + this.showIcons);
+			console.log("choiceIcons is: " + this.choiceIcons);
+			if (!this.includeIcons || !this.choiceIcons) {
+				console.log("icons not needed");
+				this.choiceIcons = this.choiceLabels;
+			}
+		}
+
+		if (this.displayMode === "Picklist") {
+			console.log("setting Picklist on");
+			this.showRadio = false;
+		}
+
+		console.log("initializing smartChoice. inputMode is: " + this.inputMode);
+		let options = [];
+		let items = [];	//parameters for visual text box selection
+		let index = 0;
+		if (this.legitInputModes.includes(this.inputMode)) {
+			switch (this.inputMode) {
+				//User can simply pass in a collection of strings as choiceValues. The same text is used for both label and value
+				case "Single String Collection":
+					console.log("entering input mode String Collection");
+					console.log("choiceValues is: " + this.choiceValues);
+					//console.log ('splitting choice values would be: ' + this.choiceValues.split(','));
+					//let values = this.choiceValues.split(';');
+
+					this.choiceValues.forEach(value => {
+						console.log("value is: " + value);
+						options.push({ label: value, value: value });
+						console.log("options is: " + options);
+					});
+					break;
+
+				//User can  pass in one collection of strings for visible labels and another for the underlying values (such as recordIds)
+				case "Dual String Collections":
+					console.log("entering input mode Dual String Collections");
+					console.log("choiceValues is: " + this.choiceValues);
+					//console.log ('splitting choice values would be: ' + this.choiceValues.split(','));
+					//let values = this.choiceValues.split(';');
+
+					this.choiceLabels.forEach(label => {
+						console.log("label is: " + label);
+						console.log("value is: " + this.choiceValues[index]);
+						options.push({ label: label, value: this.choiceValues[index] });
+						console.log("options is: " + options);
+						index += 1;
+					});
+					break;
+				
+				//User passes in Label collection of string for box header and Value collection of strings for box description
+				case "Visual Text Box":
+					console.log("entering input mode Visual Text Box");
+					console.log("choiceLabels is: " + this.choiceLabels);
+					this.choiceLabels.forEach(label => {
+						items.push({ name: label, description: this.choiceValues[index], icon: this.choiceIcons[index] });
+						console.log("items is: " + items);
+						index +=1;
+					});
+					break;
+
+				default:
+			}
+			this.options = options;
+			this.items = items;
+		} else {
+			console.log(
+				"SmartChoiceFSC: Need a valid Input Mode value. Didn't get one"
+			);
+			throw new Error(
```

Comment:
likewise, error message throws should generally be kept streamlined and one line.

Reply:
Same Vscode issue on formatting

Reply:
fwiw my vscode doesn't do that expansion so it's something that's setting-specific..... not a big deal....